### PR TITLE
Android 13 targeting

### LINKED ISSUE #2859: Improve to 10/10 on Exodus
Our assessment on Exodus is 9/10
https://reports.exodus-privacy.eu.org/fr/reports/org.kiwix.kiwixmobile/latest/

I would like to see if we can improve to 10/10

### PR DESCRIPTION
**Pull request by MohitMaliFtechiz**
Related to #2859 , let see if it will fixed it.

### CODE REVIEW

**Comment by MohitMaliFtechiz** [comment]
hi ,

We have to wait for fetch library to update before closing target android 13, We have to wait until this PR is merged into fetch https://github.com/tonyofrancis/Fetch/pull/626

**Comment by kelson42** [comment]
@MohitMaliFtechiz 
> We have to wait for fetch library to update before closing target android 13, We have to wait until this PR is merged into fetch [tonyofrancis/Fetch#626](https://github.com/tonyofrancis/Fetch/pull/626)

We can not count on this because the development of Fetch is totally stoped since years. Actually I though we use a fork of it?!

**Comment by MohitMaliFtechiz** [comment]
> 

I check fork of Fetch have not found any updated one which is avaialble on public repo store so I have override method of fetch and placed the check that support for android 13 it.

**Comment by kelson42** [comment]
OK, so this PR is ready to review? Still in draft!

**Comment by 4shutosh** [comment]
As we are moving directly from 30 to 33, we need to make sure all the API changes are considered.

I suggest you check the SplashScreenApi once and make sure the theme properties match our required behaviour.

**Comment by gouri-panda** [comment]
@MohitMaliFtechiz Thanks for this! I'll review it soon :)

File: core/src/main/java/org/kiwix/kiwixmobile/core/downloader/fetch/FetchDownloadNotificationManager.kt [hunk 1]
```diff
@@ -108,6 +122,38 @@ class FetchDownloadNotificationManager(context: Context) :
...
+  override fun getActionPendingIntent(
+    downloadNotification: DownloadNotification,
+    actionType: DownloadNotification.ActionType
+  ): PendingIntent {
+    val intent = Intent(notificationManagerAction)
+    intent.putExtra(EXTRA_NAMESPACE, downloadNotification.namespace)
+    intent.putExtra(EXTRA_DOWNLOAD_ID, downloadNotification.notificationId)
+    intent.putExtra(EXTRA_NOTIFICATION_ID, downloadNotification.notificationId)
+    intent.putExtra(EXTRA_GROUP_ACTION, false)
+    intent.putExtra(EXTRA_NOTIFICATION_GROUP_ID, downloadNotification.groupId)
```
Comment (gouri-panda) [inline core/src/main/java/org/kiwix/kiwixmobile/core/downloader/fetch/FetchDownloadNotificationManager.kt:134]: [nit] Kotlitn's `apply` operator would be nice here.

File: core/src/main/java/org/kiwix/kiwixmobile/core/downloader/fetch/FetchDownloadNotificationManager.kt [hunk 2]
```diff
@@ -108,6 +122,38 @@ class FetchDownloadNotificationManager(context: Context) :
...
+      else -> ACTION_TYPE_INVALID
+    }
+    intent.putExtra(EXTRA_ACTION_TYPE, action)
+    val flags = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+      PendingIntent.FLAG_IMMUTABLE or PendingIntent.FLAG_UPDATE_CURRENT
+    } else {
+      PendingIntent.FLAG_UPDATE_CURRENT
+    }
+    return@getActionPendingIntent PendingIntent.getBroadcast(
+      context,
```
Comment (gouri-panda) [inline core/src/main/java/org/kiwix/kiwixmobile/core/downloader/fetch/FetchDownloadNotificationManager.kt:150]: [nit] Normal `return` would be fine.

**Review by gouri-panda — APPROVED** [review]
We should also think about adding android 13 to CI.

**Review by gouri-panda — APPROVED** [review]

**Pull request merged**